design/storage-class-device-set.md

```diff
@@ -0,0 +1,493 @@
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: "failure-domain.beta.kubernetes.io/zone"
+                operator: In
+                values:
+                - us-west-1c
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 100
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                - key: "rook.io/cluster"
+                  operator: In
+                  values:
+                    - cluster1
+      volumeClaimTemplates:
+      - resources:
+          requests:
+            storage: 5Ti
+        storageClassName: gp2-ebs
+```
+
+In this example, we've added a `nodeAffinity` to the `placement` that restricts
+all OSD Pods to a specific AZ. This case is only really useful if you specify
+multiple `StorageClassDeviceSets` for different AZs, so that has been done here.
+
+#### Example 3: Different resource needs
+
+```yaml
+type: CephCluster
+name: cluster1
+...
+spec:
+  ...
+  storage:
+    ...
+    storageClassDeviceSet:
+        - name: cluster1-set1
+          count: 3
+          resources:
+            requests:
+              cpu: 2
+              memory: 4Gi
+          placement:
+            podAntiAffinity:
+              preferredDuringSchedulingIgnoredDuringExecution:
+              - weight: 100
+                podAffinityTerm:
+                  labelSelector:
+                    matchExpressions:
+                    - key: "rook.io/cluster"
+                      operator: In
+                      values:
+                        - cluster1
+                    topologyKey: "failure-domain.beta.kubernetes.io/zone"
+          volumeClaimTemplates:
+          - resources:
+              requests:
+                storage: 5Ti
+            storageClassName: gp2-ebs
+    - name: cluster1-set2
+      count: 3
+      resources:
+        requests:
+          cpu: 2
+          memory: 8Gi
+      placement:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 100
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                - key: "rook.io/cluster"
+                  operator: In
+                  values:
+                    - cluster1
+                topologyKey: "failure-domain.beta.kubernetes.io/zone"
+      volumeClaimTemplates:
+      - resources:
+          requests:
+            storage: 10Ti
+        storageClassName: gp2-ebs
+```
+
+In this example, we have two `StorageClassDeviceSets` with different capacities
+for the devices in each set. The devices with larger capacities would require a
+greater amount of memory to operate the OSD Pods, so that is reflected in the
+`resources` field.
+
+#### Example 4: Simple local storage
+
+```yaml
+type: CephCluster
+name: cluster1
+...
+spec:
+  ...
+  storage:
+    ...
+    nodes:
+    - name: node1
```

Comment:
Specifying the nodes in this list shouldn't be necessary. When the device set is created, it should match the nodes according to the `placement`. I wouldn't want to worry about matching this nodes set.

Reply:
Does Rook-Ceph not to anything to label/identify nodes that are explicitly specified in the CephCluster CR?

Reply:
Rook doesn't label nodes today? When the `nodes` are specified in the CR today, rook requires those to match the `kubernetes.io/hostname` label. 
So do you agree the `nodes` won't be used for the device sets?

Reply:
I guess I can live without it.